cli/client/filter.go

```diff
@@ -32,7 +32,7 @@ func (client *RestClient) CreateFilter(filter *types.EventFilter) (err error) {
 
 // DeleteFilter deletes a filter from configured Sensu instance
 func (client *RestClient) DeleteFilter(filter *types.EventFilter) error {
-	path := filtersPath(filter.Namespace, filter.Name)
+	path := filtersPath(client.config.Namespace(), filter.Name)
```

Comment:
I think `filter.Namespace` is correct here!

In fact this breaks the `--namespace` flag. The following works before this patch but returns `Error: not found` after applying it :(
```
sensuctl config set-namespace default
sensuctl hook delete hook1 --skip-confirm --namespace devops
```

My understanding of the problem is that since the caller of `DeleteFilter()` (here the cobra command returned by `DeleteCommand()`) doesn't set `filter.Namespace`, it gets treated as the default namespace. I think what should be done instead is set `filter.Namespace = client.config.Namespace()` in the caller of `DeleteFilter()`, and let `filter.Namespace` get overridden if the `--namespace` flag is set to a non-empty value (as it is done now).

Reply:
https://github.com/sensu/sensu-go/blob/master/cli/client/check.go#L54

Reply:
That's a good call out.. I'll have to dive a little deeper here.